lib/app_profiler/railtie.rb

```diff
@@ -60,7 +60,11 @@ class Railtie < Rails::Railtie
         if (Rails.env.development? || Rails.env.test?) && AppProfiler.stackprof_viewer.remote?
           app.middleware.insert_before(0, AppProfiler.viewer::Middleware)
         end
-        app.middleware.insert_before(0, AppProfiler.middleware)
+        if AppProfiler.otel_instrumentation_enabled
```

Comment:
Otel gems use `Rack::Events`. So inserting the profiling middleware, after `Rack::Events`, based on the config value makes sense to me.

What do you think @gmcgibbon?